sqlite3_storage_plugin/src/rosbag2/storage/sqlite/sqlite_storage.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2018,  Bosch Software Innovations GmbH.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "sqlite_storage.hpp"
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace rosbag2
+{
+
+SqliteStorage::SqliteStorage(const std::string & database_name, bool shouldInitialize)
+: database_()
+{
+  try {
+    database_ = std::make_unique<SqliteWrapper>(database_name);
+    if (shouldInitialize) {
+      initialize();
+    }
+  } catch (const SqliteException & e) {
+    throw std::runtime_error("Failed to setup storage. Error: " + std::string(e.what()));
+  }
+
+  std::cout << "Opened database '" << database_name << "'." << std::endl;
+}
+
+SqliteStorage::SqliteStorage(std::shared_ptr<SqliteWrapper> database)
+: database_(std::move(database)) {}
+
+bool SqliteStorage::write(const std::string & data)
```

Comment:
I am not sure if data here should be a `const char *` because of the string length when putting binary code in it.
Also I believe the string length is computed differently between `std::string` and `const char *`:
https://akrzemi1.wordpress.com/2014/03/20/strings-length/

Reply:
Correct. At the moment everything is hard-coded to support only string messages. This will change with https://github.com/bsinno/aos-rosbag2/issues/22.